File: edgequake/sdks/typescript/src/resources/pipeline.ts
/**
 * Pipeline resource — pipeline status and control.
 *
 * @module resources/pipeline
 * @see edgequake/crates/edgequake-api/src/handlers/pipeline.rs
 */

import type {
  CostEstimate,
  CostEstimateRequest,
  ModelPricing,
  PipelineStatus,
  QueueMetrics,
} from "../types/tasks.js";
import { Resource } from "./base.js";

export class PipelineResource extends Resource {
  /** Get pipeline processing status. */
  async status(): Promise<PipelineStatus> {
    return this._get("/api/v1/pipeline/status");
  }

  /** Cancel all running pipeline jobs. */
  async cancel(): Promise<void> {
    await this._post("/api/v1/pipeline/cancel");
  }

  /** Get queue metrics (pending, processing, failed counts). */
  async queueMetrics(): Promise<QueueMetrics> {
    return this._get("/api/v1/pipeline/queue-metrics");
  }

  /** Get model pricing information. */
  async pricing(): Promise<ModelPricing[]> {
    return this._get("/api/v1/pipeline/costs/pricing");
  }

  /** Estimate cost for a document processing request. */
  async estimateCost(request: CostEstimateRequest): Promise<CostEstimate> {
    return this._post("/api/v1/pipeline/costs/estimate", request);
  }
}

File: edgequake/sdks/rust/src/resources/pipeline.rs
//! Pipeline resource.

use crate::client::EdgeQuakeClient;
use crate::error::Result;
use crate::types::operations::*;

pub struct PipelineResource<'a> {
    pub(crate) client: &'a EdgeQuakeClient,
}

impl<'a> PipelineResource<'a> {
    /// `GET /api/v1/pipeline/status`
    pub async fn status(&self) -> Result<PipelineStatus> {
        self.client.get("/api/v1/pipeline/status").await
    }

    /// `GET /api/v1/pipeline/queue-metrics`
    pub async fn metrics(&self) -> Result<QueueMetrics> {
        self.client.get("/api/v1/pipeline/queue-metrics").await
    }

    /// `POST /api/v1/pipeline/cancel` — Cancel active pipeline.
    pub async fn cancel(&self) -> Result<()> {
        self.client
            .post_no_content::<()>("/api/v1/pipeline/cancel", None)
            .await
    }
}
